File: src/main/java/freemusic/music/service/real/WyService.java
package freemusic.music.service.real;


import com.alibaba.fastjson.JSONArray;
import com.alibaba.fastjson.JSONObject;
import freemusic.music.pojo.MainFormPojo;
import freemusic.music.pojo.wy.Ar;
import freemusic.music.pojo.wy.Wy;
import org.jetbrains.annotations.NotNull;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@Service
public class WyService implements MusicServices {

    @Value("${search.pagesize}")
    private int pagesize;

    @Override
    public ArrayList getFormatJson(@NotNull MainFormPojo form) {
        String searchResult = getSearchResult(form);
        JSONObject jsonResult = JSONObject.parseObject(searchResult).getJSONObject("result");
        if (jsonResult.getInteger("songCount") == 0) {
            return new ArrayList(0);
        }
        JSONArray jsonList = jsonResult.getJSONArray("songs");
        List<Wy> list = JSONArray.parseArray(jsonList.toJSONString(), Wy.class);
        ArrayList array = new ArrayList(list.size());
        for (Wy single : list) {
            JSONObject json = new JSONObject();
            json.put("name", single.getName());
            List<Ar> singerList = single.getAr();
            String singer = null;
            for (int a = 0; a < singerList.size(); a++) {
                singer = singerList.get(a).getName() + " ";
            }
            json.put("singer", singer);
            json.put("album", single.getAl().getName());
            json.put("s128", 0);
            json.put("sogg", 0);
            json.put("s320", single.getId());
            json.put("SQ", 0);
            int seconds = (single.getDt() / 1000) % 60;
            String secStr = seconds < 10 ? "0" + String.valueOf(seconds) : String.valueOf(seconds);
            json.put("time", (single.getDt() / 1000) / 60 + ":" + secStr);
            json.put("mv", single.getMv());
            array.add(json);
        }
        return array;
    }

    @Override
    public String getSearchResult(@NotNull MainFormPojo form) {
        String key = form.getKeyword();
        int page = Integer.valueOf(form.getPage());
        String text = "{\"s\":\"" + key + "\",\"type\":1,\"offset\":" + (page - 1) * pagesize + ",\"limit\":" + pagesize + ",\"total\":true}";
        String s = null;
        s = NetUtil.GetEncHtml("http://music.163.com/weapi/cloudsearch/get/web?csrf_token=", text, true);
        return s;
    }

}

File: src/main/java/freemusic/music/service/real/WyService.kt
package freemusic.music.service.real

import com.alibaba.fastjson.JSON
import com.alibaba.fastjson.JSONObject
import freemusic.music.pojo.MainFormPojo
import freemusic.music.pojo.wy.Ar
import freemusic.music.pojo.wy.Wy
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import java.util.ArrayList
import kotlin.collections.HashMap

@Service
class WyService : MusicServices {

    @Value("\${search.pagesize}")
    private val pageSize: Int = 0

    override fun getFormatJson(mainFormPojo: MainFormPojo): ArrayList<*> {
        val searchResult = getSearchResult(mainFormPojo)
        val jsonResult = JSONObject.parseObject(searchResult).getJSONObject("result")
        if (jsonResult.getInteger("songCount") == 0) {
            return ArrayList<Any>()
        }
        val jsonList = jsonResult.getJSONArray("songs")
        val array = ArrayList<Any>()
        for (i in jsonList.indices) {
            val single = JSON.parseObject(jsonList[i].toString(), Wy::class.java)
            val map = HashMap<String, Any>()
            map.put("name", single.name!!)
            map.put("singer",getSingers(single.ar!!))
            map.put("album", single.al!!.name!!)
            map.put("s128", "")
            map.put("sogg", "")
            map.put("s320", single.id)
            map.put("SQ", "")
            val seconds = single.dt / 1000 % 60
            val secStr = if (seconds < 10) "0" + seconds.toString() else seconds.toString()
            map.put("time", (single.dt / 1000 / 60).toString() + ":" + secStr)
            map.put("mv", single.mv)
            array.add(map)
        }
        return array
    }

    override fun getSearchResult(mainFormPojo: MainFormPojo): String {
        val key = mainFormPojo.keyword
        val page = Integer.valueOf(mainFormPojo.page)
        val text = "{\"s\":\"$key\",\"type\":1,\"offset\":" + (page - 1) * pageSize + ",\"limit\":$pageSize,\"total\":true}"
        return NetUtil.GetEncHtml("http://music.163.com/weapi/cloudsearch/get/web?csrf_token=", text, true)
    }

    private fun getSingers(singerList: List<Ar>): String {
        var singer = ""
        singerList.forEach {
            singer = it.name + " "
        }
        return singer
    }

}